### ISSUE TITLE
Problem of inconsistency of validation_bal_acc while training and the prediction with the true labels are too different.

### ISSUE THREAD

**Issue by Alex-711** [braindecode/braindecode#633]
Hi,

I experienced an issue while using the Sequence Sampler, the value of validation set balanced accuracy while training when compared with the value of balanced accuracy calculated by using the model to predict the labels are too different. 

Furthermore, the code should indicate that the loss of labels is happening only at the end of the valid set, while it should happen for each subject. In the code, there is a loss of 16 labels, and it makes sense for the valid set of 4 subjects and n_windows = 5. Yet it should be a loss of 4 labels for each subject, however it is a loss in the last 16 labels.

This code was based in the Sleep staging on the Sleep Physionet dataset using Chambon2018 network Tutorial.
Braindecode Version: 0.8.1

Copy of the code to reproduce:
https://colab.research.google.com/drive/1jitnqaBrBe-nmwAAQc3M40jbxheaVGbp?usp=sharing

If I have committed any mistakes, please contact me.

**Comment by bruAristimunha** [comment]
@tgnassou, is it related with the issue that you mentioned with sleep stage? (Cc. @agramfort)

**Comment by bruAristimunha** [comment]
Small comment, not asking you to work on this, just to see if can understand why it is happening. I have no idea.

**Comment by tgnassou** [comment]
Hi! I know that I had struggled to find the same accuracy in the training history and the prediction afterward. To fix that, I just stopped using Skorch ... I didn't find where it comes from.

For your question about the loss of labels, the problem is coming from the metadata. When you're doing `train_set.get_metadata()`, you have only one subject, but you should have 4; I don't know why. Maybe it's coming from the way of splitting `windows_dataset`; any suggestions @bruAristimunha ?

Otherwise, it's a bit weird to use sequences for BCI since each window is not continuous (separated between rest states). And I think you don't need to use balanced accuracy, the classes are already balanced.
